students/BrianE/lesson02/assignment/charges_calc.py

```diff
@@ -0,0 +1,152 @@
+
+def parse_cmd_arguments():
+    """
+    Initialize with user-provided input, output files and logging level.
+    :return: None
+    """
+
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='set logging level (1-4)', required=False, default=1)
+
+    return parser.parse_args()
+
+
+def setup_logging(level='1'):
+    """
+    Logging setup
+
+    Levels:
+                1: No debug messages or log file. (Default)
+                2: Only error messages.
+                3: Error messages and warnings.
+                4: Error messages, warnings and debug messages.
+
+    Debug:      General comments, indicating where in the script flow we are.
+                Should be shown on screen only (i.e., never saved to logfile).
+    Warning:    Used for missing elements in the source data that force a change in the flow.
+                Shown on screen and on the log file.
+    Error:      Used for inconsistencies in the source data that will cause the script
+                to crash or report incorrect results. Shown on screen and on the log file.
+
+    :param level: integer value for desired logging detail
+    :return: logger object (set with -d argument)
+    """
+
+    log_format = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+    log_file = datetime.datetime.now().strftime("%Y-%m-%d") + ".log"
+    formatter = logging.Formatter(log_format)
+
+    file_handler = logging.FileHandler(log_file)
+    file_handler.setFormatter(formatter)
+
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(formatter)
+
+    logger = logging.getLogger()
+
+    if level == '2':
+        file_handler.setLevel(logging.ERROR)
+        console_handler.setLevel(logging.ERROR)
+        logger.setLevel(logging.ERROR)
+    elif level == '3':
+        file_handler.setLevel(logging.WARNING)
+        console_handler.setLevel(logging.WARNING)
+        logger.setLevel(logging.WARNING)
+    elif level == '4':
+        file_handler.setLevel(logging.WARNING)
+        console_handler.setLevel(logging.DEBUG)
+        logger.setLevel(logging.DEBUG)
+    else:  # Default to no logging
+        file_handler.setLevel(logging.CRITICAL)
+        console_handler.setLevel(logging.CRITICAL)
+        logger.setLevel(logging.CRITICAL)
+
+    logger.addHandler(file_handler)
+    logger.addHandler(console_handler)
+
+    return logger
+
+
+def load_rentals_file(filename):
+    """
+    Read input JSON and load data as dictionary
+    :param filename: input JSON file
+    :return: data dictionary
+    """
+
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+            logger.debug(f'Successfully loaded {len(data.keys())} elements from {filename}')
+        except Exception as e:
```

Comment:
While using a generic Exception here will solve all your problems, we should try to use specific Exception catches so that the User is aware of the incorrect behavior that is causing the problem. For example, this could be because the file doesn't exist or due to the json decoder failing.